base/sparse/cholmod.jl

```diff
@@ -1242,11 +1242,7 @@ function cholfact!{Tv}(F::Factor{Tv}, A::Sparse{Tv}; shift::Real=0.0)
 end
 
 """
-    cholfact!(F::Factor, A::Union{SparseMatrixCSC{<:Real},
-        SparseMatrixCSC{Complex{<:Real}},
-        Symmetric{<:Real,SparseMatrixCSC{<:Real,SuiteSparse_long}},
-        Hermitian{Complex{<:Real},SparseMatrixCSC{Complex{<:Real},SuiteSparse_long}}};
-        shift = 0.0) -> CHOLMOD.Factor
+    cholfact!(F::Factor, A::Union{SparseMatrixCSC{<:Real},SparseMatrixCSC{Complex{<:Real}},Symmetric{<:Real,SparseMatrixCSC{<:Real,SuiteSparse_long}},Hermitian{Complex{<:Real},SparseMatrixCSC{Complex{<:Real},SuiteSparse_long}}}; shift = 0.0) -> CHOLMOD.Factor
```

Comment:
> Agree =). You also don't actually need a type alias.

Of course, but I'm not sure how to represent it otherwise...

> If people don't have any issue with the fix in this PR I'll merge this when the CI passes again. Improving the doc for those function can be another PR (I'm also not familiar with the organization of this function well enough to improve the doc myself)

This sounds trivial enough that it should be handled in this PR, instead of introducing a line wider than the guidelines recommend. Why not simply replace this type with `sparse_array`, and adapt the body of the text accordingly?